### PR TITLE
Add API Consumer ID and Type to ctx

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- Put consumer ID and Type to context.

**Related issue(s)**

### CODE REVIEW

File: components/director/internal/authenticator/consumer_ctx.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package authenticator
+
+import (
+	"context"
+	"errors"
+
+	"github.com/kyma-incubator/compass/components/director/internal/tenantmapping"
+)
+
+const ConsumerKey = "consumer type"
```
Comment (pkosiec) [inline components/director/internal/authenticator/consumer_ctx.go]: write it in camelCase

File: components/director/internal/tenantmapping/consumer.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/pkg/errors"
+)
+
+type ConsumerType string
+
+const (
+	RUNTIME            ConsumerType = "Runtime"
+	APPLICATION        ConsumerType = "Application"
+	INTEGRATION_SYSTEM ConsumerType = "Integration System"
```
Comment (pkosiec) [inline components/director/internal/tenantmapping/consumer.go]: shouldn't we use CamelCase?

File: components/director/internal/authenticator/consumer_ctx.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+var NoConsumerError = errors.New("cannot read consumer from context")
```
Comment (pkosiec) [inline components/director/internal/authenticator/consumer_ctx.go]: Think about moving this file to `internal/consumer` package, similarly to `tenant`

File: components/director/internal/tenantmapping/consumer.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package tenantmapping
+
+import (
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/pkg/errors"
+)
+
+type ConsumerType string
```
Comment (pkosiec) [inline components/director/internal/tenantmapping/consumer.go]: Similarly, this file could be in `internal/consumer` package

File: components/director/internal/tenantmapping/consumer.go (hunk 2 above)
Comment (dbadura) [reply components/director/internal/tenantmapping/consumer.go]: PascalCase in this case.

File: components/director/internal/authenticator/middleware_test.go [hunk 5]
```diff
@@ -176,15 +178,19 @@ func TestAuthenticator_Handler(t *testing.T) {
 }
 
 type jwtTokenClaims struct {
-	Scopes string `json:"scopes"`
-	Tenant string `json:"tenant"`
+	Scopes     string                `json:"scopes"`
+	Tenant     string                `json:"tenant"`
+	ObjectID   string                `json:"objectID"`
```
Comment (pkosiec) [inline components/director/internal/authenticator/middleware_test.go]: ConsumerID

File: components/director/internal/authenticator/middleware_test.go [hunk 6]
```diff
@@ -176,15 +178,19 @@ func TestAuthenticator_Handler(t *testing.T) {
...
+	ObjectType consumer.ConsumerType `json:"objectType"`
```
Comment (pkosiec) [inline components/director/internal/authenticator/middleware_test.go]: ConsumerID

**Pull request merged**